packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -608,10 +608,9 @@ export class RoutesMatcher {
 
 		let pathExistsInOutput = this.path in this.output;
 
-		// If a path with a trailing slash entered the `rewrite` phase and didn't find a match, it might
-		// be due to the `trailingSlash` setting in `next.config.js`. Therefore, we should remove the
-		// trailing slash and check again before entering the next phase.
-		if (phase === 'rewrite' && !pathExistsInOutput && this.path.endsWith('/')) {
```

Comment:
@james-elicx do you remember by any chance if there was a specific reason for the check to only apply in the `rewrite` phase?

Reply:
I think it was something to do with certain trailing slash ones being rewritten in the rewrite phase, but others missing the rewrite they need which is what this was fixing. I don't *think* it would be problematic doing it in other phases, but that would be pretty hard to tell to be honest

Reply:
yes, that's my feeling too, I think we can just go with it and see how it performs in the wild, if it becomes problematic we can always restrict on which routes this applies to

All the e2e pass and [next-apps-for-testing apps](https://github.com/dario-piotrowicz/next-apps-for-testing/actions/runs/7208472507) also seem to be totally fine (I've checked a handful of them), so it does look to me like a pretty safe change to make